tests/dhcp_relay/test_dhcp_pkt_fwd.py

```diff
@@ -0,0 +1,329 @@
+    DHCP_PKT_BOOTP_MIN_LEN = 300
+    DHCP_CLIENT = {
+        "mac": "00:11:22:33:44:55",
+        "ip": "10.10.10.1",
+        "subnet": "255.255.255.0",
+        "port": 68,
+    }
+    DHCP_RELAY = {
+        "mac": "22:33:44:55:00:11",
+        "ip": "20.20.20.1",
+        "loopback": "10.0.0.33",
+    }
+    DHCP_SERVER = {
+        "mac": "44:55:00:11:22:33",
+        "ip": "30.30.30.30",
+        "port": 67,
+    }
+
+    def __getPortLagsAndPeerIp(self, duthost, testPort):
+        """
+        Retreives all port lag members for a given testPort
+
+        Args:
+            duthost(Ansible Fixture): instance of SonicHost class of DUT
+            testPort(str): port name used for test
+
+        Returns:
+            lags(list): list of port indeces (if any) if LAG which has testPort as a member
+            peerIp(str): BGP peer IP
+        """
+        peerIp = None
+        mgFacts = duthost.minigraph_facts(host=duthost.hostname)["ansible_facts"]
+        for peer in mgFacts["minigraph_bgp"]:
+            if peer["name"] == mgFacts["minigraph_neighbors"][testPort]["name"] and \
+               ipaddr.IPAddress(peer["addr"]).version == 4:
+                peerIp = peer["addr"]
+                break
+
+        lags = [mgFacts["minigraph_port_indices"][testPort]]
+        for portchannelConfig in mgFacts["minigraph_portchannels"].values():
+            if testPort in portchannelConfig["members"]:
+                for lag in portchannelConfig["members"]:
+                    if testPort != lag:
+                        lags.append(mgFacts["minigraph_port_indices"][lag])
+                break
+
+        return lags, peerIp
+
+    def __updateRoute(self, duthost, ip, peerIp, op=""):
+        """
+        Update route to add/remove for a given IP <ip> towards BGP peer
+
+         Args:
+            duthost(Ansible Fixture): instance of SonicHost class of DUT
+            ip(str): IP to add/remove route for
+            peerIp(str): BGP peer IP
+            op(str): operation add/remove to be performed, default add
+
+        Returns:
+            None
+        """
+        logger.info("{0} route to '{1}' via '{2}'".format(
+            "Deleting" if "no" == op else "Adding",
+            ip,
+            peerIp
+        ))
+        duthost.shell("vtysh -c \"configure terminal\" -c \"{} ip route {} {}\"".format(
+            op,
+            ipaddress.ip_interface(unicode(ip + "/24")).network,
+            peerIp
+        ))
+
+    @pytest.fixture(scope="class")
+    def dutPorts(self, duthost, testbed):
+        """
+        Buils list of DUT ports and classify them as Upstream/Downstream ports.
+
+        Args:
+            duthost(Ansible Fixture): instance of SonicHost class of DUT
+            testbed(Ansible Fixture): testbed information
+
+        Returns:
+            dict: contains downstream/upstream ports information
+        """
+        if testbed["topo"]["name"] not in ("t1", "t1-lag", "t1-64-lag", "t1-64-lag-clet"):
```

Comment:
```suggestion
        if testbed["topo"]["type"] not in ("t1"):
```
Can we use the type here, or are there t1 topologies this test won't work for?

Reply:
It should work for all T1. Updated.